src/Splunk.ModularInputs/Splunk/ModularInputs/EventWriter.cs

```diff
@@ -94,11 +94,17 @@ public class EventWriter
         {
             if (completed)
                 throw new ObjectDisposedException("EventWriter already disposed.");
+
+            // Don't start the eventQueueMonitor up until we actually want to write
+            // events so we don't get empty <stream/> elements in cases where
+            // we don't actually queue anything.
             if (eventQueueMonitor == null)
-                // Don't start the eventQueueMonitor up until we actually want to write
-                // events so we don't get empty <stream/> elements in cases where
-                // we don't actually queue anything.
-                eventQueueMonitor = Task.Run(() => WriteEventsFromQueue());
+                lock (synchronizationObject)
+                {
+                    if (eventQueueMonitor == null)
```

Comment:
Is this additional null check needed? We're doing it just before acquiring the lock as well